platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformMetrics.java

```diff
@@ -137,17 +150,51 @@ public class PlatformMetrics {
 
     private final AverageAndMax avgQ1PreConsEvents;
     private final AverageAndMax avgQ2ConsEvents;
-    private final SwirldsPlatform platform;
+    private final NodeId selfId;
+    private final EventMapper eventMapper;
+    private final SwirldStateManager swirldStateManager;
+    private final CriticalQuorum criticalQuorum;
+    private final AddressBook addressBook;
+    private final PreConsensusEventHandler preConsensusEventHandler;
+    private final ConsensusRoundHandler consensusRoundHandler;
+    private final SimultaneousSyncThrottle simultaneousSyncThrottle;
+    private final EventStreamManager<EventImpl> eventStreamManager;
+    private final StateManagementComponent stateManagementComponent;
 
     /**
      * Constructor of {@code PlatformMetrics}
      *
-     * @param platform
-     * 		a reference to the {@link SwirldsPlatform}
      */
-    public PlatformMetrics(final SwirldsPlatform platform) {
-        this.platform = CommonUtils.throwArgNull(platform, "platform");
-        final Metrics metrics = platform.getContext().getMetrics();
+    public PlatformMetrics(
+            @NonNull final Metrics metrics,
+            @NonNull final NodeId selfId,
+            @NonNull final FreezeManager freezeManager,
+            @NonNull final StartUpEventFrozenManager startUpEventFrozenManager,
+            @Nullable final SyncManagerImpl syncManager,
+            @NonNull final EventMapper eventMapper,
+            @NonNull final SwirldStateManager swirldStateManager,
+            @NonNull final CriticalQuorum criticalQuorum,
+            @NonNull final AddressBook addressBook,
```

Comment:
I do not think that this goes in a good direction. We should add an inversion of control. Maybe `platform.initMetrics(PlatformMetrics m)`? Adding @netopyr